src/eddymotion/utils.py

```diff
@@ -0,0 +1,176 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# We support and encourage derived works from this project, please read
+# about our expectations at
+#
+#     https://www.nipreps.org/community/licensing/
+#
+"""Utils to sort the DWI data volume indices"""
+
+from enum import Enum
+
+import numpy as np
+
+
+class SortingStrategy(Enum):
+    """
+    Enum class representing different sorting strategies.
+
+    Available sorting strategies:
+    - LINEAR: Sorts the items in a linear order.
+    - RANDOM: Sorts the items in a random order.
+    - BVALUE: Sorts the items based on their b-value.
+    - CENTRALSYM: Sorts the items based on their central symmetry.
+    """
+
+    LINEAR = "linear"
+    RANDOM = "random"
+    BVALUE = "bvalue"
+    CENTRALSYM = "centralsym"
+
+
+def sort_dwdata_indices(dwdata, strategy, seed=None):
+    """Sort the DWI data volume indices following the given strategy.
+
+    Parameters
+    ----------
+    dwdata : :obj:`~eddymotion.dmri.DWI`
+        DWI dataset, represented by this tool's internal type.
+    strategy : :obj:`~eddymotion.utils.SortingStrategy`
+        The sorting strategy to be used. Available options are:
+        - SortingStrategy.LINEAR: Sort the indices linearly.
+        - SortingStrategy.RANDOM: Sort the indices randomly.
+        - SortingStrategy.BVALUE: Sort the indices based on the last column of gradients in ascending order.
+        - SortingStrategy.CENTRALSYM: Sort the indices in a central symmetric manner.
+    seed : :obj:`int` or :obj:`bool`, optional
+        Seed the random number generator. If an integer, the value is used to
+        initialize the generator; if ``True``, the arbitrary value
+        of ``20210324`` is used to initialize it.
+
+    Returns
+    -------
+    index_order : :obj:`numpy.ndarray`
+        The sorted index order.
+    """
+    if strategy == SortingStrategy.LINEAR:
+        return linear_action(dwdata)
+    elif strategy == SortingStrategy.RANDOM:
+        return random_action(dwdata, seed)
+    elif strategy == SortingStrategy.BVALUE:
+        return bvalue_action(dwdata)
+    elif strategy == SortingStrategy.CENTRALSYM:
+        return centralsym_action(dwdata)
+    else:
+        raise ValueError("Invalid sorting strategy")
+
+
+def linear_action(dwdata):
+    """
+    Sort the DWI data volume indices linearly
+
+    Parameters
+    ----------
+    dwdata : :obj:`~eddymotion.dmri.DWI` DWI dataset, represented by this tool's internal type.
+
+    Returns
+    -------
+    index_order : :obj:`numpy.ndarray`
+    The sorted index order.
```

Comment:
Comment lost after new commits:
```suggestion
    Parameters
    ----------
    size : :obj:`int`
        Number of volumes in the dataset
        (for instance, the number of orientations in a DWI)

    Returns
    -------
    :obj:`list` of :obj:`int`
        The sorted index order.

    Examples
    --------
    >>> linear_action(10)
    [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]

```